### PR TITLE
Expose list of build assets under `/assets/-/:pkg/:env/:version?`

### PR DESCRIPTION
**Pull request by indexzero**
Follow-up from our discussion today @jcrugzz:

1. `GET /assets/-/:pkg/:env/:version?` is unauthenticated. This is safe because the `files` within any `Build` is public by definition. 
2. Deduplicates code from `GET /builds/:pkg/:env/:version?` into `app.spec.{fromRequest,findBuild}` which is used in both locations. This route should exist and should remain authenticated since the other build metadata is **not** public by definition.

#### Before merging:

- [x] Integration tests
- [x] Define filtering semantics
- [x] Discuss if an _"I'm feeling lucky"_ style route is useful with a particular HTTP encoding (i.e. returning the first file in the set of files as text).

### CODE REVIEW

File: lib/routes/assets.js [hunk 1]
```diff
@@ -14,6 +14,38 @@ var comment = require('commenting'),
...
+
+        //
+        // TODO (indexzero): proper sanitization on this untrusted
+        // user input.
+        //
+        const { filter } = req.query;
+        if (filter) {
+          res.send(
+            build.files.filter(file => {
+              return path.extname(file) === filter;
```
Comment (jcrugzz) [inline lib/routes/assets.js]: should we limit this to just extension? if we are we should support `.js` and `js` so with or without the `.`. My thought is that we could make it string includes based but idk if thats too complicated

Comment (indexzero) [reply lib/routes/assets.js]: Having `filter` be the extension name where the leading `.` is stripped automatically was my intention.

File: lib/preboots/spec.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
+
```
Comment (indexzero) [inline lib/preboots/spec.js]: Add JSDoc here.

File: lib/routes/builds.js [hunk 3]
```diff
@@ -43,6 +24,24 @@ module.exports = function (app) {
 
   });
 
+  app.routes.get('/builds/-/meta/:pkg/:version', auth, function (req, res, next) {
+    var error;
+
+    app.bffs.meta(spec(req.params, req.query), function meta(err, build) {
```
Comment (jcrugzz) [inline lib/routes/builds.js]: this actually needs to be `app.bffs.meta(spec.fromRequest(req)...`

**Pull request merged**